mewarpx/mewarpx/appendablearray.py

```diff
@@ -0,0 +1,242 @@
+"""Array type which can be appended to in an efficient way.
+"""
+__all__ = ['AppendableArray','DynamicHistogram','DynamicHistogramIntersect']
+import sys
+import numpy
+import types
+# Class which allows an appendable array.
+# DPG 8/19/99
+
+
+class AppendableArray:
+    """
+  Creates an array which can be appended to in an efficient manner. The object
+  keeps an internal array which is bigger than the actual data. When new data is
+  appended, it is only copied to fill in the extra space. More space is only
+  allocated when that space fills up. This saves both the allocation time, and
+  the time to copy the existing data to the new space.
+   - initlen=1: The initial size of the array. The most efficiency is gained
+                when initlen is close to the total expected length.
+   - unitshape=None: The appendable unit can be an array. This gives the shape of
+                 the unit. The full shape of the array then would be
+                 [n]+unitshape, where n is the number of units appended.
+   - typecode='i': Typecode of the array. Uses the same default as the standard
+                   array creation routines.
+   - autobump=100: The size of the increment used when additional extra space
+                   is needed.
+   - initunit=None: When given, the unitshape and the typecode are taken from
+                    it. Also, this unit is make the first unit in the array.
+   - aggressivebumping=1.5: Whenever new space is added, autobump will be
+                            increased by this factor in an attempt to
+                            minimize the number of reallocations. Its new
+                            value will be the max of the size of the appended data
+                            and this times the old autobump size. A good value is
+                            1.5 - this can greatly reduce the amount of
+                            rallocation without a significant amount of wasted
+                            space.
+  Create an instance like so
+  >>> a = AppendableArray(initlen=100,typecode='d')
+  Append a single unit like this
+  >>> a.append(7.)
+  or multiple units like this
+  >>> a.append(numpy.ones(5,'d'))
+  The data can be obtained by directly indexing the instance, like this
+  >>> print a[:4]
+  [ 7., 1., 1., 1.,]
+  will give the first four number appended
+  Other methods include len, data, setautobump, cleardata, reshape
```

Comment:
Update doc string format so it looks like what we currently use.
<paragraph>
Arguments:
<paragraph>